_posts/2023-06-15-user-survey-2023.md

```diff
@@ -0,0 +1,240 @@
+Thankfully, the results we got match what we were expecting:
+
+The overwhelming majority of our users runs Linux, typically Red Hat Enterprise Linux or one of its derivatives, which tends to be most common in industry settings, or Debian or Ubuntu, which is used widely in "less managed" settings, as well as in CI.
+
+Even though the cocotb project has spent a very significant amount of effort in engineering and support on it, our Windows user base remains comparably small.
+
+<div class="col">
+  <img src="/assets/blog/2023-06-15-user-survey-2023/8-os-distro.svg" class="img-fluid mx-auto d-block" alt="operating system or Linux distribution">
+</div>
+
+Cocotb aims to support as many Python versions as possible, including versions which have been marked as "end-of-life" by the Python project itself (such as Python 3.6).
+Each new Python version brings significant benefits, with the most recent versions also being substantially faster in many use cases.
+We are therefore happy to see many users relying on recent or very recent versions of Python!
+
+Many things can go wrong when one tries to compile Python by itself, as opposed to using a (RPM or deb) package prepared by a Linux distribution, conda, or python.org.
+We are therefore equally happy to see that the large majority of users makes use of these packages.
+
+<div class="row p-5">
+  <div class="col">
+    <img src="/assets/blog/2023-06-15-user-survey-2023/11-python-version.svg" class="img-fluid mx-auto d-block" alt="Python installation source">
+  </div>
+  <div class="col">
+    <img src="/assets/blog/2023-06-15-user-survey-2023/7-install-python.svg" class="img-fluid mx-auto d-block" alt="Python installation source">
+  </div>
+</div>
+
+cocotb pushes a new release roughly twice a year, and takes great care in ensuring backwards-compatibility to allow our users to switch to the latest version.
+The survey results confirm that this effort is paying off: 83 percent of users are on the latest 1.7 release (at the time of the survey).
+
+<img src="/assets/blog/2023-06-15-user-survey-2023/10-cocotb-version.svg" class="img-fluid mx-auto d-block p-5" alt="cocotb version">
+
+A look at the simulators used with cocotb completes the picture.
+When choosing proprietary simulators, Cadence Xcelium and Siemens ModelSim and Questa lead the field.
+In the world of open source simulators, Verilator and Icarus Verilog are the most popular choices.
+Icarus Verilog is the default simulator in many cocotb tutorials and even its Makefiles, so its popularity is hardly surprising.
+Equally, GHDL is the most mature open source VHDL simulator, hence being the go-to choice for VHDL developers.
+
+What is surprising, however, is the amount of people using Verilator together with cocotb.
+Verilator is an excellent simulator, however, it lacks a number of features that make it integrate nicely with cocotb.
+It seems that these limitations are not severe enough to prevent its use with cocotb for many users.
```

Comment:
Ehhhh, I might leave this out. It is a judgement of someone not in the room. It does *work* now, it just behaves different than other simulators.

Reply:
Well, it does not pass the cocotb test suite, so for the definition of cocotb, Verilator doesn't fully work. But happy to leave this out.